### PR TITLE
fix(throttle): throttle globally by default and optionally per route

### PR DESCRIPTION
**Pull request by simylein**

### CODE REVIEW

File: lib/core/boot/boot.ts [hunk 1]
```diff
@@ -71,32 +71,62 @@ export const bootstrap = (): FluxifyServer => {
...
-							entry.hits += 1;
-							if (entry.hits > throttle.limit) {
-								debug(`throttle limit on route ${endpoint}`);
-								const status = 429;
-								stop(request, 'throttle');
-								return createResponse({ status, message: 'too many requests' }, status, request, {
-									'retry-after': `${Math.ceil((entry.exp - Date.now()) / 1000)}`,
-								});
+						if (config.throttleTtl === throttle.ttl && config.throttleLimit === throttle.limit) {
+							const glob = global.server.throttle[request.ip]?.['']?.[''];
```
Comment (slowlydev) [inline lib/core/boot/boot.ts:75]: maybe call it `globalThrottle` as glob is weird and global is taken?

**Review by slowlydev — CHANGES_REQUESTED** [review]
Is there a way to turn throttling off globally? (I probably missed it)

**Comment by simylein** [comment]
> Is there a way to turn throttling off globally? (I probably missed it)

It is still turned off as it needs to be enabled explicitly by configuring it in the environment. 
This only affects you when you enable throttling, we throttle on a per ip per route basis before. 
With this change we throttle on a per ip basis and additionally if configured on a by route basis. 

Example:
global throttle limit 16 => you can call 16 different routes or one route 16 times. 
global throttle limit 16 and /slow is 4 => you have a budget of 16 requests but on route /slow you only got 4.

File: lib/core/boot/boot.ts (hunk 1 above)
Comment (simylein) [reply lib/core/boot/boot.ts:75]: Yeah global is taken because environment defines it. I need to rename entry too...

**Review by slowlydev — APPROVED** [review]
Okay lgtm

**Pull request merged**